section segmentation is not 100% reproducible

While running tests that test segmentation for jun 21st, we found that the tests fail intermittently.

The error is

```
======================================================================
FAIL: testJun21 (analysisTests.intakeTests.TestPipelineRealData.TestPipelineRealData)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "<https://amplab.cs.berkeley.edu/jenkins/job/e-mission-server-prb/ws/emission/tests/analysisTests/intakeTests/TestPipelineRealData.py",> line 126, in testJun21
    ad.AttrDict(ground_truth).data)
  File "<https://amplab.cs.berkeley.edu/jenkins/job/e-mission-server-prb/ws/emission/tests/analysisTests/intakeTests/TestPipelineRealData.py",> line 60, in compare_result
    self.assertEqual(rs.properties.enter_fmt_time, es.properties.enter_fmt_time)
AssertionError: u'2016-06-21T15:44:22.801000-07:00' != u'2016-06-21T15:34:16.888000-07:00'
- 2016-06-21T15:44:22.801000-07:00
?               ^  ^^  ^^
+ 2016-06-21T15:34:16.888000-07:00
?               ^  ^^  ^^
```

All the places are fine, so the overall trips is fine.
This is only an issue with section segmentation.

```
        "feature_type": "start_place", 
        "exit_fmt_time": "2016-06-21T15:24:10.399000-07:00", 
```

```
        "feature_type": "start_place", 
        "exit_fmt_time": "2016-06-21T15:24:10.399000-07:00", 
```

```
        "enter_fmt_time": "2016-06-21T15:54:37.544000-07:00", 
        "feature_type": "end_place", 
```

```
        "enter_fmt_time": "2016-06-21T15:54:37.544000-07:00", 
        "feature_type": "end_place", 
```

In the ground truth, we have a very short stop at 15:34 that is missing from
the subsequent run. Note that this is the very first stop.

```
2016-08-18 15:30:43,205:DEBUG:140735159468032:------- Comparing trip feature ---------
        "enter_fmt_time": "2016-06-21T15:44:22.801000-07:00", 
        "feature_type": "stop", 
        "exit_fmt_time": "2016-06-21T15:44:41.301000-07:00", 

        "enter_fmt_time": "2016-06-21T15:34:16.888000-07:00", 
        "feature_type": "stop", 
        "exit_fmt_time": "2016-06-21T15:34:57.738000-07:00", 

AssertionError: u'2016-06-21T15:44:22.801000-07:00' != u'2016-06-21T15:34:16.888000-07:00'
- 2016-06-21T15:44:22.801000-07:00
```

The next stop in the ground truth matches up with the first one in the new run.

```
                    "properties": {
                        "enter_fmt_time": "2016-06-21T15:44:22.801000-07:00",
                        "feature_type": "stop",
                        "exit_fmt_time": "2016-06-21T15:44:41.301000-07:00",
```

So for some reason, the ground truth has an extra stop that is missing from the test case.
I don't have logs for the ground truth, so looking at the logs for the test case now.
I will also try to re-run and see if I can reproduce the ground truth run...

--- a/emission/storage/decorations/trip_queries.py
+++ b/emission/storage/decorations/trip_queries.py
@@ -18,10 +18,10 @@ def get_cleaned_sections_for_trip(user_id, trip_id):
     return get_sections_for_trip("analysis/cleaned_section", user_id, trip_id)
 
 def get_raw_stops_for_trip(user_id, trip_id):
-    return get_sections_for_trip("segmentation/raw_stop", user_id, trip_id)
+    return get_stops_for_trip("segmentation/raw_stop", user_id, trip_id)
 
 def get_cleaned_stops_for_trip(user_id, trip_id):
-    return get_sections_for_trip("analysis/cleaned_stop", user_id, trip_id)
+    return get_stops_for_trip("analysis/cleaned_stop", user_id, trip_id)
 
 def get_raw_timeline_for_trip(user_id, trip_id):
     """
@@ -46,9 +46,9 @@ def get_sections_for_trip(key, user_id, trip_id):
     """
     query = {"user_id": user_id, "data.trip_id": trip_id,
              "metadata.key": key}
+    logging.debug("About to execute query %s with sort_key %s" % (query, "data.start_ts"))
     section_doc_cursor = edb.get_analysis_timeseries_db().find(query).sort(
         "data.start_ts", pymongo.ASCENDING)
-    logging.debug("About to execute query %s" % query)
     return [ecwe.Entry(doc) for doc in section_doc_cursor]
 
 def get_stops_for_trip(key, user_id, trip_id):
@@ -57,7 +57,7 @@ def get_stops_for_trip(key, user_id, trip_id):
     """
     query = {"user_id": user_id, "data.trip_id": trip_id,
              "metadata.key": key}
-    logging.debug("About to execute query %s" % query)
+    logging.debug("About to execute query %s with sort_key %s" % (query, "data.enter_ts"))
     stop_doc_cursor = edb.get_analysis_timeseries_db().find(query).sort(
         "data.enter_ts", pymongo.ASCENDING)
     return [ecwe.Entry(doc) for doc in stop_doc_cursor]

